camelot/view/action_steps/crud.py

```diff
@@ -108,14 +112,21 @@ class SetColumns(ActionStep):
     
     def __init__(self, static_field_attributes):
         self.static_field_attributes = static_field_attributes
+        self.column_attributes = []
+        for fa in static_field_attributes:
+            attrs = {}
+            if fa['delegate'].__name__ == 'ComboBoxDelegate':
+                attrs['action_routes'] = fa['action_routes']
+            self.column_attributes.append(attrs)
 
     def _to_dict(self):
         columns = []
-        for fa in self.static_field_attributes:
+        for i, fa in enumerate(self.static_field_attributes):
             columns.append({
                 'verbose_name': str(fa['name']),
                 'field_name': fa['field_name'],
                 'width': fa['column_width'],
+                'delegate': fa['delegate'](**self.column_attributes[i])._to_dict(),
```

Comment:
hier moet je nu toch geen delegate constureren, en kan je toch gewoon de column attributes rechtstreeks er in steken ?

Reply:
Het is een named dataclass, we hebben dus ook die name nodig. Hier is mss een [...__name__, ...] wel efficiënter zonder dat er type fouten gebeuren.